examples/tree_2d_dgsem/elixir_euler_kelvin_helmholtz_instability_ethz.jl

```diff
@@ -0,0 +1,108 @@
+using OrdinaryDiffEq
+using Trixi
+using StableRNGs
+
+
+###############################################################################
+# semidiscretization of the compressible Euler equations
+gamma = 1.4
+equations = CompressibleEulerEquations2D(gamma)
+
+"""
+    initial_condition_kelvin_helmholtz_instability_ethz(x, t, equations::CompressibleEulerEquations2D)
+
+A version of the classical Kelvin-Helmholtz instability based on
+- Ulrik S. Fjordholm, Roger Käppeli, Siddhartha Mishra, Eitan Tadmor (2014)
+  Construction of approximate entropy measure valued
+  solutions for hyperbolic systems of conservation laws
+  [arXiv: 1402.0909](https://arxiv.org/abs/1402.0909)
+"""
+function initial_condition_kelvin_helmholtz_instability_ethz(x, t, equations::CompressibleEulerEquations2D)
+  # typical resolution 128^2, 256^2
+  # doomain size is [0,+1]^2
+  # interface is sharp, but randomly perturbed
+  rng = StableRNG(100)
+  m = 10
+  a1 = rand(rng,m)
+  a2 = rand(rng,m)
+  a1 = a1./sum(a1)
+  a2 = a2./sum(a2)
+  b1 = (rand(rng,m).-0.5) * pi
+  b2 = (rand(rng,m).-0.5) * pi
+  Y1 = 0.0
+  Y2 = 0.0
+  for n = 1:m
+    Y1 += a1[n] * cos(b1[n] + 2 * n * pi * x[1])
+    Y2 += a2[n] * cos(b2[n] + 2 * n * pi * x[1])
+  end
+
+  J1 = 0.25
+  J2 = 0.75
+  epsilon = 0.01
+  I1 = J1 + epsilon * Y1
+  I2 = J2 + epsilon * Y2
```

Comment:
This looks like the correct implementation from the paper to me. If it were crucial for performance, we could avoid additional allocations by moving the construction of the random parameters out of the function but inside a `let` block for type stability. Alternatively, we could declare this as a callable struct. Anyway, I think it's not relevant for performance here, so it's okay :+1:

Reply:
Thanks for the tip. I also checked the PID before making the PR and I get about 1.5e-7s for the new KHI setup and about the same number for the standard KHI setup with the same discretization. So I had the feeling it should be fine...